sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -115,6 +115,36 @@ def _forward_pass(self, activations):
 
         return activations
 
+    def _forward_pass_fast(self, X):
+        """Predict using the trained model
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            The input data.
+
+        Returns
+        -------
+        y_pred : ndarray of shape (n_samples,) or (n_samples, n_outputs)
+            The decision function of the samples for each class in the model.
+        """
+        X = check_array(X, accept_sparse=['csr', 'csc'])
+
+        # Initialize first layer
+        activation = X
+
+        # Forward propagate
+        hidden_activation = ACTIVATIONS[self.activation]
+        for i in range(self.n_layers_ - 1):
+            activation = safe_sparse_dot(activation, self.coefs_[i])
+            activation += self.intercepts_[i]
+            if i != self.n_layers_ - 2:
```

Comment:
Nit but for consistency with the other version:

```suggestion
            if (i + 1) != (self.n_layers_ - 1):
```

Reply:
The original version uses `i + 1` several times, but this simplified version only uses `i`, so it's more clear here to not have the extra addition in the if statement.